### ISSUE TITLE
[Bug]: Podman machine fails to start with exit status 255 on Mac

### ISSUE THREAD

**Issue by smakinen** [containers/podman#17403]
### Issue Description

There seems to be an issue when trying to start Podman with `podman machine start` when using macOS and QEMU. I created a Podman machine about two months ago but now the machine fails to start. Somehow, starting the machine got gradually worse after a time before completely failing in startup.

Here is what happens when I try to start the machine.

```
podman machine start 
Starting machine "podman-machine-default"
Waiting for VM ...
Mounting volume... /Users/me:/Users/me
Error: exit status 255
```

From the `podman machine start --log-level debug` output, I can see the last statement that is executed is the SSH command related to creating the mount point directories (maybe line 669 in qemu/machine.go), which fails. Previous issues have considered this to be a sign of e.g. invalid SSH configuration (e.g. [https://github.com/containers/podman/issues/14237](https://github.com/containers/podman/issues/14237)) but maybe there is something more to it. When running with `log-level debug` and the QEMU window open, the exit status 255 error shows before all the Fedora services have started.

Could this be a race condition issue so that SSH related services have not yet started when the SSH mount commands are executed? I found one closed and apparently fixed issue where the race condition was suggested (https://github.com/containers/podman/issues/11532).

I tried connecting to the QEMU QMP monitor socket with `nc -U qmp_podman-machine-default.sock` and running the following queries just after QEMU has started and before all the services are running.

```json
{"QMP": {"version": {"qemu": {"micro": 0, "minor": 2, "major": 7}, "package": ""}, "capabilities": ["oob"]}}
{ "execute": "qmp_capabilities"} 
{"return": {}}
{ "execute": "query-status" }
{"return": {"status": "running", "singlestep": false, "running": true}}
```
So the VM is in an running state and also the gvproxy port (50810 here) is in a listening state early on.

```
netstat -ap tcp | grep -i listen
tcp4       0      0  localhost.50810        *.*                    LISTEN
```
Is it possible that condition in qemu/machine.go (line 645) `for state != machine.Running || !listening { ` cannot hold back the execution of the SSH statement before the machine is fully initialized?

### Steps to reproduce the issue

Steps to reproduce the issue (happens on an existing Podman machine)  
1. `podman machine start`  

### Describe the results you received

```
podman machine start 
Starting machine "podman-machine-default"
Waiting for VM ...
Mounting volume... /Users/me:/Users/me
Error: exit status 255
```

### Describe the results you expected

Podman machine should be able to start. Thinking more broadly, maybe there should additional guarantees that SSH on the QEMU machine is up and running before issuing commands to the machine. Perhaps the SSH connection should be polled for a couple of times with sensible timeout or then other events from QEMU used for the purpose? For instance, the QMP monitor emits a NIC_RX_FILTER_CHANGED event towards the end of the initialization (not sure of its purpose, though).

```json
{"timestamp": {"seconds": 1675763180, "microseconds": 921089}, "event": "NIC_RX_FILTER_CHANGED", "data": {"path": "/machine/peripheral-anon/device[0]/virtio-backend"}}
```

### podman info output

The machine (or gvproxy) is not up and running so cannot get info.

```
podman info   
Error: failed to connect: dial tcp [::1]:50810: connect: connection refused
```

Some supplemental environment details to `podman info` .

```yaml
Podman (brew): stable 4.4.0 (bottled), HEAD
OS: macOS Ventura 13.2
Architecture (QEMU): aarch64
Chip: Apple M1 Pro
```

### Podman in a container

No

### Privileged Or Rootless

Rootless

### Upstream Latest Release

Yes

### Additional environment details

Here are the results of `podman machine inspect`.

```json
[
  {
    "ConfigPath": {
      "Path": "/Users/me/.config/containers/podman/machine/qemu/podman-machine-default.json"
    },
    "ConnectionInfo": {
      "PodmanSocket": {
        "Path": "/Users/me/.local/share/containers/podman/machine/podman-machine-default/podman.sock"
      }
    },
    "Created": "2022-12-12T12:34:18.127475+02:00",
    "Image": {
      "IgnitionFilePath": {
        "Path": "/Users/me/.config/containers/podman/machine/qemu/podman-machine-default.ign"
      },
      "ImageStream": "testing",
      "ImagePath": {
        "Path": "/Users/me/.local/share/containers/podman/machine/qemu/podman-machine-default_fedora-coreos-37.20221127.2.0-qemu.aarch64.qcow2"
      }
    },
    "LastUp": "2023-01-31T16:41:24.124649+02:00",
    "Name": "podman-machine-default",
    "Resources": {
      "CPUs": 1,
      "DiskSize": 100,
      "Memory": 3072
    },
    "SSHConfig": {
      "IdentityPath": "/Users/me/.ssh/podman-machine-default",
      "Port": 50810,
      "RemoteUsername": "core"
    },
    "State": ""
  }
]
```

### Additional information

It appears that the problem affects Podman machines that have been in use for some time (i.e. a few months). Why fresh  Podman machines straight out-of-the-box work and what cause the slow decay remain a mystery.

**Comment by smakinen** [comment]
Here is what I have tried so far without success:

- upgraded QEMU from version 7.1.0 to 7.2.0 (brew)
- upgraded Podman from version 4.3.1 to 4.4.0 (brew)
- upgraded macOS from Monterey to Ventura 13.2
- in QEMU configuration podman-machine-default.json, verified parameters -cpu host and -M virt ([https://github.com/containers/podman/issues/14303](https://github.com/containers/podman/issues/14303), [https://github.com/containers/podman/pull/14563](https://github.com/containers/podman/pull/14563))  
- in QEMU configuration podman-machine-default.json, increased VM memory to 3072 MB
- in SSH configuration, added `IdentitiesOnly yes` to localhost ([https://github.com/containers/podman/issues/14237](https://github.com/containers/podman/issues/14237)).

Is recreating the machine the only way forward at this point? I guess there is no way to recover containers from the VM if the machine cannot be connected?

**Comment by jamesmikesell** [comment]
I see this error quite frequently too.

Not sure if this is related, but feel like i started seeing this error more frequently after starting to use Podman for [dev containers](https://containers.dev/), and additionally needing to add swap space to the `podman machine`.

**Comment by jamesmikesell** [comment]
@smakinen I think you are correct about this being a race condition, and some bug internal to Podman

I'm able to get podman to start a machine that was previously failing using the following shell script, which manually pauses the `podman` and `gvproxy` process (no idea what that second command does), and waits until the user verifies that the Qemu VM has finished booting before unpausing those processes.


```shell
#!/bin/bash

pkill podman
pkill qemu

podman machine start --log-level debug &
PID=$!
sleep 2

pkill -STOP podman
pkill -STOP gvproxy

echo "^^^^"
echo "^^^^"
echo "^^^^"
echo "If the above says the VM already running or started"
echo "then edit the json file located at ~/.config/containers/podman/machine/qemu/"
echo "and change the line"
echo "\"Starting\": true"
echo "to be"
echo "\"Starting\": false"
echo ""
echo "dont forget to save, and rerun this script."
echo ""
echo "Else, continue with instructions below"
echo ""
echo "Qemu will open in another window (likely in the background)"
echo "wait until you see a login prompt on that window"
read -p "then return to THIS terminal and hit enter"

pkill -CONT podman
pkill -CONT gvproxy
```


Unfortunately, this is only a bandaid.  We need something to address whatever the underlying root problem is.

**Comment by tonyseek** [comment]
It happened on an Intel MacBook (MacBookPro16,1) too, not just the M1. @jamesmikesell's script works for me, as a workaround.

**Comment by berndlosert** [comment]
I just started having this problem after restarting my M1 Mac. Running with --log-level debug shows that it dies after trying to ssh into the VM.

**Comment by berndlosert** [comment]
Removing the vm and recreating it fixed it for me.

**Comment by smakinen** [comment]
A brilliant idea and move @jamesmikesell! I'm happy to say that with your script, I can see my containers again :). So halting the podman and gvproxy processes until QEMU has time to finish initialization works. I actually previously tried to renice the QEMU and Podman processes to make the QEMU process run faster but I could not figure out how to halt the other process altogether. Thanks a lot.

To @berndlosert: also good to hear that recreating the VM works but I did not want lose my VM since I had a bunch of containers in the VM used for development. So I was not keen to remove the existing VM (and the problem could also reappear after a while).

For Podman and the QEMU machine initialization code, I think there should a function such as _isSSHRunning_ which would be used to check the SSH status before using SSH (in the 'Waiting for VM...' phase). It's a good question what is the best way to check the availability of SSH in the guest OS if the machine state and the listening-state of gvproxy cannot be relied on.

**Comment by laurent-martin** [comment]
I observed the same, podman vm starting successfully only once every 4 times.
but pausing podman and gvproxy until qemu is stable solves the situation. cool, thanks.

**Comment by benoitf** [comment]
I reproduce it time to time as well with the latest version
```
$ podman machine start new
Starting machine "new"
Waiting for VM ...
Mounting volume... /Users:/Users
Error: exit status 255
```

**Comment by barloff-st** [comment]
Hitting same issue with the following:

- A1990 (2019 Intel)
- podman 4.3.1 and 4.4.4 (upgrade did not help)

Workarounds mentioned here did not resolve the issue (removing VM and recreating, and script).  Running into same failure location as what **berndlosert** is seeing.

**Comment by itsthejoker** [comment]
I just ran into this issue as well with podman 4.4.4 on a 2020 MBP. `podman machine stop / start` a few times let it finish, but I'm seeing the same results as @benoitf.

**Comment by smakinen** [comment]
For me, James' script has worked well so far. Waiting until QEMU is prepared for SSH logins helps. I'm certain that the waiting approach could be forged into a PR for Podman but perhaps it won't work in all cases. I'm a bit curious, what do you @itsthejoker and others see in the **QEMU window** when you run `podman machine start --log-level debug` and wait until no more messages are printed to the QEMU screen? Is there some kind of other error message perhaps related to SSH?

**Comment by chevdor** [comment]
I kept on having issue and I am seeing some success thanks to [this comment](https://github.com/containers/podman/issues/14237#issuecomment-1298270596) in [this issue](https://github.com/containers/podman/issues/14237). I think the topic of this issue could be edited and the M1 part removed, I don't think the issues discussed here are M1 specific. I run into the same on an Intel Mac. The issue is however likely related to the fact that users here use a Mac.

The podman troubleshooting guide mentions some extra cleanup so the following may help some:
```
podman machine stop
podman machine rm -f
rm -rf ~/.local/share/containers/podman
rm -rf ~/.config/containers/
```

then your regular:
```
podman machine init
podman machine start
```

AFAIK, the fix for me was to add the following:
```
# to fix podman issues
# see https://github.com/containers/podman/issues/14237
Host localhost
  IdentitiesOnly yes
```
to my `~/.ssh/config`.

I was consistently getting this `255` error even from a clean rebooted system and when using a default podman machine.

With the fix above of the ssh config and yet-another cleanup, I was able to finally get things back in order.

**Comment by samuel-phan** [comment]
For me, the issue was my SSH agent with a lot of keys already.

My fix:

```bash
ssh-add -D  # clear the SSH keys from my SSH agent

podman machine stop
podman machine start
```

**Comment by noyez** [comment]
> For me, the issue was my SSH agent with a lot of keys already.

Same here, i just ran into this issue. I just temporarily unset SSH_AUTH_SOCK. 

@samuel-phan didn't know about `ssh-add -D`, thx!

**Comment by paulftw** [comment]
> see the script I use [start.sh](https://github.com/laurent-martin/podman_x86_64_on_apple_aach64/blob/main/start.sh)

Had the same issue and the start script fixed it. I also had to change "Starting" to false in json (as @jamesmikesell mentioned). Without it any attempt to start a machine was returning "VM already running or starting".

**Comment by laurent-martin** [comment]
> I also had to change "Starting" to false in json

I have updated the script here: https://github.com/laurent-martin/podman_x86_64_on_apple_aach64
So that it automatically fixes this dondition as well now:

```text
./podmac.sh start intel_64
Checking jq...OK, jq found
Checking podman...OK, podman found
Checking curl...OK, curl found
Resetting stale starting state.         <--------------------
Starting machine "intel_64"
Waiting for VM ...
Pausing podman
SSH available: Resuming podman.
Mounting volume... /Users/laurent:/Users/laurent

This machine is currently configured in rootless mode. If your containers
require root permissions (e.g. ports < 1024), or if you run into compatibility
issues with non-podman clients, you can switch using the following command:

	podman machine set --rootful intel_64

API forwarding listening on: /var/run/docker.sock
Docker API clients default to this address. You do not need to set DOCKER_HOST.

Machine "intel_64" started successfully
```

**Comment by vrothberg** [comment]
@ashley-cui @baude PTAL

**Comment by osalbahr** [comment]
This happens to me sometimes too, even though the machine starts anyways. I don't know if this is related.
```
~$ podman machine start
Starting machine "podman-machine-default"
Waiting for VM ...
Mounting volume... /Users:/Users
Error: exit status 255
~$ podman run -it debian
Resolved "debian" as an alias (/etc/containers/registries.conf.d/000-shortnames.conf)
Trying to pull docker.io/library/debian:latest...
Getting image source signatures
Copying blob sha256:918547b9432687b1e1d238e82dc1e0ea0b736aafbf3c402eea98c6db81a9cb65
Copying config sha256:34b4fa67dc04381e908b662ed69b3dbe8015fa723a746c66cc870a5330520981
Writing manifest to image destination
Storing signatures
root@1c153be897f5:/# uname -a
Linux 1c153be897f5 6.2.9-300.fc38.x86_64 #1 SMP PREEMPT_DYNAMIC Thu Mar 30 22:32:58 UTC 2023 x86_64 GNU/Linux
root@1c153be897f5:/# exit
exit
~$ 
```

**Comment by laurent-martin** [comment]
> Mounting volume... /Users:/Users
> Error: exit status 255
> ~$ podman run -it debian

Yes, the VM may be started, but the mount failed. The test is rather (/Users represents the mounted volume in VM):

```bash
$ podman run -it -v /Users:/Users debian
root@a1f31cce35f5:/# ls /Users
```

It should show the contents of /Users on macos... when the mount did not fail.

**Comment by osalbahr** [comment]
> > Mounting volume... /Users:/Users
> > Error: exit status 255
> > ~$ podman run -it debian
> 
> Yes, the VM may be started, but the mount failed. The test is rather (/Users represents the mounted volume in VM):
> 
> ```shell
> $ podman run -it -v /Users:/Users debian
> root@a1f31cce35f5:/# ls /Users
> ```
> 
> It should show the contents of /Users on macos... when the mount did not fail.

I am aware that the mount didn’t fail unlike with OP. I was just reporting on the error message in case it might be related to this issue, since it is part of the same message.

**Comment by jamesmikesell** [comment]
I've come up with a slightly improved version of my script (read hack) above which, which waits for the VM to finish booting before letting podman mount directories.

The above script ran podman/qemu in debug mode, which slowed down performance of the running containers.

This improved version of the script runs podman/qemu in normal mode, and relies on the user waiting for the CPU utilization of the `qemu` process to lower and stabilize (something that happens once the VM is done loading) before hitting `enter` and thus allowing podman to mount shared directories.

```shell
#!/bin/bash

pkill podman
pkill qemu

podman machine start &
PID=$!
sleep 2

pkill -STOP podman
pkill -STOP gvproxy

echo "^^^^"
echo "^^^^"
echo "^^^^"
echo "If the above says the VM already running or started"
echo "then edit the json file located at ~/.config/containers/podman/machine/qemu/"
echo "and change the line"
echo "\"Starting\": true"
echo "to be"
echo "\"Starting\": false"
echo ""
echo "don't forget to save, and rerun this script."
echo ""
echo "Else, continue with instructions below"
echo ""
echo "Wait until the displayed CPU utilization lowers and stabilizes to 1% or less"
echo "Then hit enter"


PID_QEMU=$(pgrep qemu)

while true; do
  CPU=$(ps -p $PID_QEMU -o %cpu | awk 'NR>1 {print $1}')
  printf "\rCPU utilization: %s%%             " $CPU

  read -s -t 1
  if [ $? -eq 0 ]; then
    break
  fi
done


pkill -CONT podman
pkill -CONT gvproxy
```

**Comment by ashley-cui** [comment]
Took a look at this and I'm still struggling to find a correct solution - This error code definitely comes from a [failed SSH call](https://github.com/containers/podman/blob/03578810759b35e9f15e4452a29d8667f1fcced1/pkg/machine/qemu/machine.go#L688) during the while mounting directories into the VM during the start.  The solutions above indicate that it's a race condition, however, we do [check](https://github.com/containers/podman/blob/03578810759b35e9f15e4452a29d8667f1fcced1/pkg/machine/qemu/machine.go#LL663C3-L663C3) that the SSH socket is up before doing the call, and in my testing, the failure happens even if the socket ping succeeds, which shouldn't be the case, as the successful ping tells us that the SSH daemon is listening on the socket. 

I've tried adding a 1 second sleep, and also putting the SSH portion into a for loop to re-try on failure, but the sleep doesn't seem to help, as error still persists, and putting the SSH call to re-try on failure seems to run into another problem: sometimes after a failed mount, QEMU hangs and isn't able to be properly stopped using Podman. 

I'm not entirely sure what else to check here, but I'm leaving my findings here if anyone wants to take this up, as I'm out for the next week. cc @mheon @baude

**Comment by laurent-martin** [comment]
> I've tried adding a 1 second sleep, and also putting the SSH portion into a for loop to re-try on failure

Hi,
did you try this script:
https://github.com/laurent-martin/podman_x86_64_on_apple_aach64

It is similar to @jamesmikesell script, but it automatically detect SSH availability in the VM.

**Comment by smakinen** [comment]
> The solutions above indicate that it's a race condition, however, we do [check](https://github.com/containers/podman/blob/03578810759b35e9f15e4452a29d8667f1fcced1/pkg/machine/qemu/machine.go#LL663C3-L663C3) that the SSH socket is up before doing the call, and in my testing, the failure happens even if the socket ping succeeds, which shouldn't be the case, as the successful ping tells us that the SSH daemon is listening on the socket.

Thanks @ashley-cui for checking out the problem. I think the current method of pinging the SSH port might indeed be the problem. If Go's net.DialTimeout just checks the listening state of the local SSH gvproxy port, it might not mean that the SSH daemon on the Fedora guest OS is also running. So possibly isListening() != 'Fedora SSH services up and running'. That's what makes the problem tricky. 

The known solutions provided in this thread work around the issue in different ways. The script of @jamesmikesell uses an indirect metric of CPU utilization of the qemu process and waits for user input. In the solution provided by @laurent-martin, the availability of SSH is checked by listening to the SSH daemon's opening statement (grepping the SSH string which is something like _SSH-2.0-OpenSSH_8.8_). To see how late in the process SSH is actually ready, you can use the nc command found in Laurent's solution on the Mac host (enter the command some time after starting the QEMU process) and wait for some time.

`nc localhost <localhost_gvproxy_port>`

Having said this, I hope that the suggested and linked PR works but it might also not if it still relies on the old isListening code as such.

**Comment by baude** [comment]
@smakinen when you observe this failure, can you also verify if gvproxy is running or not?  I've picked up this issue and I am currently honing in on a fix.  Like @ashley-cui , I very much struggle to replicate this issue so I have had to depend on others to get answers and give me access to their machine.

Right now, I believe that all of these reports are really the combination of more than one bug ... one of which seems to always be this timing issue.

For the timing issue, I have been experimenting with altering the "ready" service in the machine to trigger later (like after NM).  I may also make it trigger even later, like sometime around when zincati fires which is a pretty good indicator we are actually up and running.

**Comment by smakinen** [comment]
> @smakinen when you observe this failure, can you also verify if gvproxy is running or not? I've picked up this issue and I am currently honing in on a fix. Like @ashley-cui , I very much struggle to replicate this issue so I have had to depend on others to get answers and give me access to their machine.

Yeah, I can imagine it's not so easy to find a machine where Podman does not start. If it's any consolation, the reproduction rate is pretty high if you do happen to come by a machine in this state. Answering @baude to your question, yeah, gvproxy is running in the background early on. Gvproxy is up and the port is in a listening state almost instantly after executing `podman machine start` and seeing the text 'Waiting for VM ...' on the screen. Of course, the gvproxy process dies at the same time when podman exists with the error code.

**Comment by hsoong** [comment]
Hope to fix this BIG BUG as soon as possible.

**Comment by vrothberg** [comment]
After https://github.com/containers/podman/pull/19060, I see this issue much less frequently.  In fact, today I did not manage reproducing it all.

**Comment by vrothberg** [comment]
One more observation: `machine ssh` does not suffer from the flake.  I can run that in a tight loop for 20 minutes with no flake.

**Comment by cgwalters** [comment]
I think almost every tool out there doing ssh ends up doing retries, FWIW.

Only skimming this I do agree with Dusty it's likely `systemd-user-sessions.service` being asynchronous with ssh.  One reason for this is I think is to support logging in as `root` for recovery scenarios, as distinct from "users".

For FCOS we certainly promote `core@` as the admin user though.

In the end I think podman-machine could inject Ignition that added a drop-in for `After=systemd-user-sessions.service` for `sshd.service` to close the race.

For others, it might be more of a feature though.

**Comment by vrothberg** [comment]
Thanks for confirming, @cgwalters!

I agree and opened https://github.com/containers/podman/pull/19210.

**Comment by romanrev** [comment]
I am still experiencing this issue - on `podman` 4.5.1 - using the `podmac.sh`, removing all of my ssh identities etc - have to start/stop machine for around 20 times before it actually mounts the directories correctly
```
➤ podmac.sh start podman-machine-default
Checking jq...OK, jq found
Checking podman...OK, podman found
Checking curl...OK, curl found
Starting machine "podman-machine-default"
Waiting for VM ...
Pausing podman
Waiting for SSH /...Mounting volume... /Users:/Users
Error: exit status 255
ERROR: podman exited prematurely
roman@Romans-MacBook-Pro:~/K/A/m/d/c/application|main⚡*
➤ podman machine stop
Waiting for VM to exit...
Machine "podman-machine-default" stopped successfully
# above in a loop
# until finally

➤ podmac.sh start podman-machine-default
Checking jq...OK, jq found
Checking podman...OK, podman found
Checking curl...OK, curl found
Starting machine "podman-machine-default"
Waiting for VM ...
Pausing podman
Waiting for SSH /...Mounting volume... /Users:/Users
SSH available: Resuming podman.
Mounting volume... /private:/private
Mounting volume... /var/folders:/var/folders
API forwarding listening on: /var/run/docker.sock
Docker API clients default to this address. You do not need to set DOCKER_HOST.
```

**Comment by Stausssi** [comment]
The bug is not fixed in 4.5.1, but will be part of the 4.6.0 release (see changelog)

**Comment by vrothberg** [comment]
Yes, that's it.  The fixes will ship with the upcoming 4.6 release.  The plan is to release 4.6 by the end of this week, so I the fix will reach you soon.

**Comment by itsthejoker** [comment]
Awesome, thank you!!

**Comment by sneko** [comment]
Can someone confirm v4.6 solves the issue? In my case I upgraded 1 week ago, had the issue multiple times despite a full reboot of my MacOS...

https://github.com/containers/podman/issues/17403#issuecomment-1536636874 is still my only viable solution 😢

**Comment by benoitf** [comment]
@sneko did you recreate your podman machine ? Ignition script has been changed so you'll need to recreate the machine

**Comment by romanrev** [comment]
> Can someone confirm v4.6 solves the issue? In my case I upgraded 1 week ago, had the issue multiple times despite a full reboot of my MacOS...
> 
> [#17403 (comment)](https://github.com/containers/podman/issues/17403#issuecomment-1536636874) is still my only viable solution 😢

I seem to have stopped experiencing the issue after I upgraded, even without recreating the machine

**Comment by smakinen** [comment]
Thanks for the changes to the Ignition script and for shipping the fix. For a previously created Podman machine that was failing in machine startup, I still got the exit status 255 error with Podman 4.6.0. As Florent mentioned, taking advantage of the changed Ignition script won't happen without recreating the machine.

`% podman --version    
podman version 4.6.0
% podman machine start
Starting machine "podman-machine-default"
Waiting for VM ...
Error: machine did not transition into running state: ssh error: exit status 255`

I think the situation is fairly good now. If all future machines start ok and at least some of the currently failing legacy machines can be started with the delayed startup scripts found here, Podman should be good to go in most scenarios 😊. Thank you all and have a nice autumn (at least in the northern hemisphere) everyone 🍁.

**Comment by laurent-martin** [comment]
Starting a x86 VM on M1 macbook works now with 4.6.0 : 

```console
% podman --version
podman version 4.6.0

% podman machine start intel_64
Starting machine "intel_64"
Waiting for VM ...
Mounting volume... /Users/laurent:/Users/laurent

This machine is currently configured in rootless mode. If your containers
require root permissions (e.g. ports < 1024), or if you run into compatibility
issues with non-podman clients, you can switch using the following command:

	podman machine set --rootful intel_64

API forwarding listening on: /var/run/docker.sock
Docker API clients default to this address. You do not need to set DOCKER_HOST.

Machine "intel_64" started successfully
```

**Comment by shrishs** [comment]
I am still facing the same issue on Mac Ventura.
podman version
Client:       Podman Engine
Version:      4.6.1


bash-3.2$ podman machine start podman-machine-default --log-level debug
INFO[0000] podman filtering at log level debug          
DEBU[0000] Using Podman machine with `qemu` virtualization provider 
Starting machine "podman-machine-default"
[/usr/local/opt/podman/libexec/podman/gvproxy -listen-qemu unix:///var/folders/ft/1tqtxfq54v38g9kbzvby4hkm0000gn/T/podman/qmp_podman-machine-default.sock -pid-file /var/folders/ft/1tqtxfq54v38g9kbzvby4hkm0000gn/T/podman/podman-machine-default_proxy.pid -ssh-port 53958 -forward-sock /Users/shrishsrivastava/.local/share/containers/podman/machine/qemu/podman.sock -forward-dest /run/user/501/podman/podman.sock -forward-user core -forward-identity /Users/xxxx/.ssh/podman-machine-default --debug]
DEBU[0000] qemu cmd: [/usr/local/bin/qemu-system-x86_64 -m 2048 -smp 1 -fw_cfg name=opt/com.coreos/config,file=/Users/xxxx/.config/containers/podman/machine/qemu/podman-machine-default.ign -qmp unix:/var/folders/ft/1tqtxfq54v38g9kbzvby4hkm0000gn/T/podman/qmp_podman-machine-default.sock,server=on,wait=off -netdev socket,id=vlan,fd=3 -device virtio-net-pci,netdev=vlan,mac=5a:94:ef:e4:0c:ee -device virtio-serial -chardev socket,path=/var/folders/ft/1tqtxfq54v38g9kbzvby4hkm0000gn/T/podman/podman-machine-default_ready.sock,server=on,wait=off,id=apodman-machine-default_ready -device virtserialport,chardev=apodman-machine-default_ready,name=org.fedoraproject.port.0 -pidfile /var/folders/ft/1tqtxfq54v38g9kbzvby4hkm0000gn/T/podman/podman-machine-default_vm.pid -machine q35,accel=hvf:tcg -cpu host -virtfs local,path=/Users,mount_tag=vol0,security_model=none -virtfs local,path=/private,mount_tag=vol1,security_model=none -virtfs local,path=/var/folders,mount_tag=vol2,security_model=none -drive if=virtio,file=/Users/xxxx/.local/share/containers/podman/machine/qemu/podman-machine-default_fedora-coreos-38.20230806.2.0-qemu.x86_64.qcow2] 
Waiting for VM ...
Error: machine did not transition into running state

**Comment by vrothberg** [comment]
@shrishs thanks for reporting.  I don´t think it's the same issue but another issue.  Feel free to create a new issue on GitHub.
